src/ncbitaxon.py

```diff
@@ -141,7 +152,9 @@ def convert_node(node, label, merged, synonyms, citations):
     rank = node["rank"]
     if rank and rank != "" and rank != "no rank":
         if rank not in ranks:
-            print(f"WARN Unrecognized rank '{rank}'")
+            if rank not in UNRECOGNIZED_RANKS:
+                print(f"unrecognized rank: '{rank}'")
+            UNRECOGNIZED_RANKS[rank] += 1
```

Comment:
Instead of using number, you can update the counter this way `UNRECOGNIZED_RANKS.update([rank])`

Reply:
I tested, and it worked as expected.

Reply:
I think the issue here is I've violated single responsibility principle. We aren't going to see any logging based on this change since this PR also adds the configuration to make this logging go away

Reply:
I've removed the changes related to logging from this PR